src/components/SWActivationCard/SWActivationCard.js

```diff
@@ -0,0 +1,406 @@
+    const {
+      switchAccount, accounts, navigation, gasInfo, assets,
+    } = this.props;
+    const { deployEstimateFee, gasEstimate } = this.state;
+    const keyBasedAccount = accounts.find(account => account.type === ACCOUNT_TYPES.KEY_BASED);
+    if (!keyBasedAccount) return;
+    await switchAccount(keyBasedAccount.id);
+
+    const gasPriceWei = getGasPriceWei(gasInfo);
+    const gasPrice = gasPriceWei.toNumber();
+    const assetsArray = getAssetsAsList(assets);
+    const asset: any = assetsArray.find((_asset: any) => _asset.name === 'Ethereum');
+    const {
+      symbol,
+      address: contractAddress,
+      decimals,
+    } = asset;
+
+
+    navigation.navigate(SMART_WALLET_UNLOCK, {
+      transferTransactions: [{
+        gasLimit: gasEstimate,
+        gasPrice,
+        symbol,
+        contractAddress,
+        decimals,
+        amount: parseFloat(utils.formatEther(deployEstimateFee.toString())),
+      }],
+    });
+  }
+
+  activateSW = () => {
+    const { selectedWallet } = this.state;
+    const { deploySmartWallet } = this.props;
+    if (selectedWallet === ACCOUNT_TYPES.SMART_WALLET) {
+      deploySmartWallet();
+    } else {
+      this.deployFromLegacyWallet();
+    }
+    this.setState({ isModalVisible: false });
+  }
+
+  renderModal = () => {
+    const { balances, accounts, theme } = this.props;
+    const { isModalVisible, selectedWallet } = this.state;
+
+    const legacyWallet = accounts.find(acc => acc.type === ACCOUNT_TYPES.KEY_BASED);
+    const smartWallet = accounts.find(acc => acc.type === ACCOUNT_TYPES.SMART_WALLET);
+
+    const ethBalanceInKeyWallet = legacyWallet && getBalance(balances[legacyWallet.id], ETH);
+    const ethBalanceInSmartWallet = smartWallet && getBalance(balances[smartWallet.id], ETH);
+
+    const transactionEstimate = this.getTransactionEstimate();
+
+    const buttonEnabled = selectedWallet === ACCOUNT_TYPES.SMART_WALLET ?
+      (ethBalanceInSmartWallet && ethBalanceInSmartWallet > transactionEstimate) :
+      (ethBalanceInKeyWallet && ethBalanceInKeyWallet > transactionEstimate);
+
+    const themeType = getThemeType(theme);
+
+
+    return (
+      <SlideModal
+        isVisible={isModalVisible}
+        onModalHide={() => {
+          this.setState({ isModalVisible: false });
+        }}
+        hideHeader
+      >
+        <SafeAreaView>
+          <ModalContainer>
+            <MediumText center medium>Activate Smart Wallet</MediumText>
+            <Spacing h={18} />
+            <CachedImage
+              style={{ width: 64, height: 64, alignSelf: 'center' }}
+              source={themeType === DARK_THEME ? smartWalletIconDark : smartWalletIcon}
+            />
+            <Spacing h={20} />
+            <BaseText medium>
+              Enable better security and free instant transactions via Pillar Network. You will get a new badge too.
+            </BaseText>
+            <Spacing h={15} />
+            {!!ethBalanceInSmartWallet && (
```

Comment:
What if I just created my smart wallet and don't have any eth here, but have in my key based wallet?
We should display that case, disable activation from the smart wallet (since it's empty) and preselect the key based option